### PR TITLE
Disabled -Xfatal-warnings outside of CI

### PR DESCRIPTION
**Pull request by MarconZet**

### CODE REVIEW

**Comment by MarconZet** [comment]
On CI:
```
[info] welcome to sbt 1.5.2 (Ubuntu Java 11.0.9.1)
[info] loading settings for project akka-serialization-helper-build from plugins.sbt ...
[info] loading project definition from /home/circleci/akka-serialization-helper/project
[info] loading settings for project akka-serialization-helper from build.sbt ...
[info] resolving key references (10313 settings) ...
[info] set current project to akka-serialization-helper (in build file:/home/circleci/akka-serialization-helper/)
[info] * -deprecation
[info] * -encoding
[info] * UTF-8
[info] * -feature
[info] * -language:_
[info] * -Xlog-reflective-calls
[info] * -Xlint:_
[info] * -Ybackend-parallelism
[info] * 8
[info] * -Ywarn-dead-code
[info] * -Ywarn-unused:-imports,_
[info] * -unchecked
[info] * -Xfatal-warnings
[success] Total time: 0 s, completed Jun 30, 2021, 1:58:13 PM
```

File: build.sbt [hunk 1]
```diff
@@ -41,14 +41,14 @@ lazy val borerAkkaSerializer = (projectMatrix in file("borer-akka-serializer"))
...
     libraryDependencies ++= Seq(
-        borerCore,
-        akkaTyped % Provided,
-        reflections,
-        borerDerivation % Test,
-        akkaTestKit % Test,
-        akkaStream % Test,
-        akkaStreamTestKit % Test,
-        enumeratum % Test))
+      borerCore,
```
Comment (PawelLipski) [inline build.sbt:44]: Whoops it keeps shuttling there and back... `sbt scalafmtSbtCheck`? See Hydra's Jenkinsfile, stage `Check formatting & code style`

Comment (PawelLipski) [reply build.sbt:44]: Plus project/.scalafmt.conf etc.

File: build.sbt [hunk 2]
```diff
@@ -19,19 +19,19 @@ Global / onChangedBuildSource := ReloadOnSourceChanges
...
+    "-feature",
+    "-language:_",
+    "-Xlog-reflective-calls",
+    "-Xlint:_",
+    "-Ybackend-parallelism",
+    "8",
+    "-Ywarn-dead-code",
+    "-Ywarn-unused:-imports,_",
+    "-unchecked",
+    if (sys.env.getOrElse("CI", "false") == "true") "-Xfatal-warnings" else ""),
```
Comment (PawelLipski) [inline build.sbt:34]: Hmm aren't there any problems from an empty-string argument?

Comment (MarconZet) [reply build.sbt:34]: I tested locally with `sbt clean compile test` and it exited with `0`

File: build.sbt (hunk 1 above)
Comment (MarconZet) [reply build.sbt:44]: Ok, I figured out what was happening. It seems my IDE is using different settings for formatting. It's a local fix.

**Review by PawelLipski — APPROVED** [review]

**Pull request merged**